File: web-nodejs/extension-common/src/model/FlexSearchOption.ts
import {FlexSearchOptionItem} from "./FlexSearchOptionItem";

/**
 * 搜索的关键字的 key 和可选的过滤项目
 */
export interface FlexSearchOption {
    /**
     * 搜索关键字的 key
     */
    keywordKey: string | undefined;

    /**
     * 搜索选项
     */
    options: FlexSearchOptionItem[] | undefined;
}

File: lib-extension/src/main/java/com/xiaoyv/flexiflix/extension/model/FlexSearchOption.kt
package com.xiaoyv.flexiflix.extension.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

/**
 * 搜索的关键字的 key 和可选的过滤项目
 */
@Parcelize
data class FlexSearchOption(
    @SerializedName("keywordKey") var keywordKey: String,
    @SerializedName("options") var options: List<FlexSearchOptionItem>? = null
) : Parcelable

